deployment/ccip/changeset/v1_6/cs_ccip_home.go

```diff
@@ -1401,3 +1414,127 @@ func ValidateCCIPHomeConfigSetUp(
 	}
 	return nil
 }
+
+type DeployDonIDClaimerConfig struct {
+	HomeChainSelector uint64 `json:"homeChainSelector"`
+}
+
+func DeployDonIDClaimerChangeset(e deployment.Environment, cfg DeployDonIDClaimerConfig) (deployment.ChangesetOutput, error) {
+	state, err := changeset.LoadOnchainState(e)
+	if err != nil {
+		e.Logger.Errorw("Failed to load existing onchain state", "err", err)
+		return deployment.ChangesetOutput{}, err
+	}
+
+	if err := cfg.Validate(e, state); err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("%w: %w", deployment.ErrInvalidConfig, err)
+	}
+
+	ab := deployment.NewMemoryAddressBook()
+	chain := e.Chains[cfg.HomeChainSelector]
+	err = deployDonIDClaimerContract(e, ab, state, chain)
+	if err != nil {
+		e.Logger.Errorw("Failed to deploy donIDClaimer contract", "err", err, "addressBook", ab)
+		return deployment.ChangesetOutput{
+			AddressBook: ab,
+		}, fmt.Errorf("failed to deploy donIDClaimer contract: %w", err)
+	}
+	return deployment.ChangesetOutput{
+		Proposals:   []timelock.MCMSWithTimelockProposal{},
+		AddressBook: ab,
+	}, nil
+}
+
+func deployDonIDClaimerContract(e deployment.Environment, ab deployment.AddressBook, state changeset.CCIPOnChainState, chain deployment.Chain) error {
+	chainState, chainExists := state.Chains[chain.Selector]
+	if !chainExists {
+		return fmt.Errorf("chain %s not found in existing state, deploy the prerequisites first", chain.String())
+	}
+
+	if state.Chains[chain.Selector].DonIDClaimer == nil {
+		_, err := deployment.DeployContract(e.Logger, chain, ab,
+			func(chain deployment.Chain) deployment.ContractDeploy[*don_id_claimer.DonIDClaimer] {
+				donIDClaimerAddr, tx2, donIDClaimerC, err2 := don_id_claimer.DeployDonIDClaimer(
+					chain.DeployerKey,
+					chain.Client,
+					chainState.CapabilityRegistry.Address(),
+				)
+				return deployment.ContractDeploy[*don_id_claimer.DonIDClaimer]{
+					Address: donIDClaimerAddr, Contract: donIDClaimerC, Tx: tx2, Tv: deployment.NewTypeAndVersion(changeset.DonIDClaimer, deployment.Version1_6_1), Err: err2,
+				}
+			})
+		if err != nil {
+			e.Logger.Errorw("Failed to deploy donIDClaimer contract", "chain", chain.String(), "err", err)
+			return err
+		}
+	} else {
+		e.Logger.Infow("DonIDClaimer already deployed", "chain", chain.String(), "addr", chainState.DonIDClaimer.Address)
+	}
+
+	return nil
+}
+
+func (d DeployDonIDClaimerConfig) Validate(e deployment.Environment, state changeset.CCIPOnChainState) error {
+	return donIDClaimerValidationHelper(state, d.HomeChainSelector)
+}
+
+type DonIDClaimerOffSetConfig struct {
+	HomeChainSelector uint64 `json:"homeChainSelector"`
+	OffSet            uint32 `json:"offset"`
+}
+
+func DonIDClaimerOffSetChangeset(e deployment.Environment, cfg DonIDClaimerOffSetConfig) (deployment.ChangesetOutput, error) {
```

Comment:
We should be using the new changeset API, see how the consolidated changesets are built using precondition and logic functions.